### PR TITLE
Search borrow

### PR DESCRIPTION
**Pull request by at-duongtruong**
## Description
A few sentences describing the overall goals of the pull request's commits.
## Reference Requirements
- Product Backlog ID: PBI-4
- Task ID: #4.2

## Migrations
YES

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)
select count(*) as aggregate from `borrows` left join `books` on `books`.`id` = `borrows`.`book_id` left join `users` on `users`.`id` = `borrows`.`user_id` where `name` LIKE '%Ms%' and `borrows`.`status` = '0'

select `users`.`employ_code`, `users`.`name`, `users`.`email`, `books`.`title`, `borrows`.`from_date`, `borrows`.`to_date`, `borrows`.`id` from `borrows` left join `books` on `books`.`id` = `borrows`.`book_id` left join `users` on `users`.`id` = `borrows`.`user_id` where `name` LIKE '%Ms%' and `borrows`.`status` = '0' order by `from_date` desc limit 10 offset 0

select count(*) as aggregate from `borrows` left join `books` on `books`.`id` = `borrows`.`book_id` left join `users` on `users`.`id` = `borrows`.`user_id` where `title` LIKE '%Mr%' and `borrows`.`status` = '0'

select `users`.`employ_code`, `users`.`name`, `users`.`email`, `books`.`title`, `borrows`.`from_date`, `borrows`.`to_date`, `borrows`.`id` from `borrows` left join `books` on `books`.`id` = `borrows`.`book_id` left join `users` on `users`.`id` = `borrows`.`user_id` where `title` LIKE '%Mr%' and `borrows`.`status` = '0' order by `from_date` desc limit 10 offset 0

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

File: resources/lang/en/borrows.php [hunk 1]
```diff
@@ -13,6 +13,6 @@
     'search' => 'Search',
     'list_search' =>[
         \App\Model\Borrow::TYPE_NAME => 'Name',
-        \App\Model\Borrow::TYPE_BORROWING => 'Book Borrowing',
+        \App\Model\Borrow::TYPE_TITLE => 'Book Borrowing',
     ],
 ];
```
Comment (at-trampham) [inline resources/lang/en/borrows.php]: LGTM

File: app/Http/Controllers/Admin/BorrowController.php [hunk 2]
```diff
@@ -24,9 +24,8 @@ public function index()
             'borrows.to_date',
             'borrows.id',
         ];
-        $borrows = Borrow::select($fields)
-        ->join('users', 'users.id', '=', 'borrows.user_id')
-        ->join('books', 'books.id', '=', 'borrows.book_id')
+        $borrows = Borrow::search(request('search'), request('filter'))
+        ->select($fields)
```
Comment (at-huenguyen) [inline app/Http/Controllers/Admin/BorrowController.php]: lùi vào 4 spaces nữa nha em

**Review by ghost — CHANGES_REQUESTED** [review]

File: app/Model/Borrow.php [hunk 3]
```diff
@@ -73,4 +75,20 @@ public function books()
     {
         return $this->belongsTo(Book::class, 'book_id');
     }
+
+    /**
+     * The attributes that can be search.
+     *
+     * @var array $searchableFields
+     */
+    protected $searchableFields = [
```
Comment (at-huenguyen) [inline app/Model/Borrow.php]: em tạo ra cái này để làm gì? và em dùng ở đâu vại?

**Review by ghost — CHANGES_REQUESTED** [review]
Em dùng đầu mà lắm câu query ở phần description vại, trong khi trong code thì có 1 câu truy vấn mà

**Review by ghost — APPROVED** [review]

**Pull request merged**